### PR TITLE
Specify keystore secret

### LINKED ISSUE #7472: Feature request: incubator/elasticsearch needs a way to put secure settings in a keystore
FEATURE REQUEST

  - Describe *in detail* the feature/behavior/change you'd like to see.
I want to be able to set secure settings in elasticsearch.  This can't be done in the regular config file that we can append from the configmap.  

**Version of Helm and Kubernetes**:
Helm 2.9.1 on Kubernetes 1.10

**Which chart**: incubator/elasticsearch

**What happened**:Could not set a secure setting from the chart

**What you expected to happen**:to be able to add a secure setting to the keystore file in the elasticsearch 

**Anything else we need to know**:
I've had a conversation with folks who work on the elasticsearch docker image here about how to customize the container (https://github.com/elastic/elasticsearch-docker/issues/189). With respect to secure settings, they recommended mounting in a secret to the key-store file.  I agree and will probably post a patch soon.

### PR DESCRIPTION
**Pull request by yokhahn**
@simonswine @icereval @rendhalver

**What this PR does / why we need it**:
This allows a keystore to be mounted into place so that you can set secure settings in elasticsearch

**Which issue this PR fixes**  fixes #7472

**Special notes for your reviewer**:

### CODE REVIEW

File: incubator/elasticsearch/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: elasticsearch
 home: https://www.elastic.co/products/elasticsearch
-version: 1.5.0
+version: 1.5.1
```
Comment (paulczar) [inline incubator/elasticsearch/Chart.yaml]: This is adding functionality and should therefor be a minor update which would make this version `1.6.0`.

File: incubator/elasticsearch/values.yaml [hunk 2]
```diff
@@ -17,6 +17,9 @@ cluster:
   # If you want X-Pack installed, switch to an image that includes it, enable this option and toggle the features you want
   # enabled in the environment variables outlined in the README
   xpackEnable: false
+  # Some settings must be placed in a keystore, so they need to be mounted in from a secret.
+  # Use this setting to specify the name of the secret
+  # keystoreSecret: eskeystore
```
Comment (paulczar) [inline incubator/elasticsearch/values.yaml:22]: this new variable should also be added to the README.md

**Review by paulczar — CHANGES_REQUESTED** [review]

File: incubator/elasticsearch/Chart.yaml (hunk 1 above)
Comment (cpanato) [reply incubator/elasticsearch/Chart.yaml]: @yokhahn need to bump the chart version to `1.6.0` per @paulczar comments

File: incubator/elasticsearch/values.yaml (hunk 2 above)
Comment (cpanato) [reply incubator/elasticsearch/values.yaml:22]: still missing this in README.md

Comment (yokhahn) [reply incubator/elasticsearch/values.yaml:22]: just added it

**Comment by cpanato** [comment]
@yokhahn you need bump the chart version

**Comment by cpanato** [comment]
@yokhahn you need to sign the DCO https://github.com/helm/charts/pull/7473/checks?check_run_id=15110402

File: incubator/elasticsearch/Chart.yaml (hunk 1 above)
Comment (yokhahn) [reply incubator/elasticsearch/Chart.yaml]: Should be now

**Comment by yokhahn** [comment]
hmmm.. the rebase for adding the DCO signoff went badly... because there was a merge in there.

**Comment by yokhahn** [comment]
I'm going to close this and have another go

**Pull request closed without merging**